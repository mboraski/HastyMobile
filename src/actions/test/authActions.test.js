import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import {
    signInWithFacebook,
    signInWithEmailAndPassword,
    signUp,
    signOut,
    updateAccount
} from '../authActions';

jest.mock('expo', () => ({
    Facebook: {
        logInWithReadPermissionsAsync: jest.fn(
            () =>
                new Promise((resolve, reject) => {
                    resolve({ token: 'token' });
                })
        )
    }
}));

describe('Testing Auth Actions', () => {
    const middlewares = [thunk];
    const mockStore = configureStore(middlewares);
    const initialState = {};
    const store = mockStore(initialState);

    beforeEach(() => {
        store.clearActions();
    });

    it('signInWithFacebook', async () => {
        await store.dispatch(signInWithFacebook());
        const actions = store.getActions();
        expect(actions).toMatchSnapshot();
    });

    it('handles signInWithEmailAndPassword action', async () => {
        await store.dispatch(signInWithEmailAndPassword({ email: 'email', password: 'password' }));
        const actions = store.getActions();
        expect(actions).toMatchSnapshot();
    });
    it('handles signUp action', async () => {
        await store.dispatch(
            signUp({ email: 'email', password: 'password', name: 'name', number: '1112223333' })
        );
        const actions = store.getActions();
        expect(actions).toMatchSnapshot();
    });
    it('handles signOut action', async () => {
        await store.dispatch(signOut());
        const actions = store.getActions();
        expect(actions).toMatchSnapshot();
    });
    it('handles updateAccount action', async () => {
        await store.dispatch(updateAccount({ name: 'name', number: '1112223333' }));
        const actions = store.getActions();
        expect(actions).toMatchSnapshot();
    });
});
