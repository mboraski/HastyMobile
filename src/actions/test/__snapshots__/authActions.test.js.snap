// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Auth Actions facebookLogin 1`] = `
Array [
  Object {
    "payload": Object {
      "token": "token",
    },
    "type": "login_success",
  },
  Object {
    "payload": "token",
    "type": "login_user",
  },
]
`;
